Assets/Scripts/Runtime/GameManager.cs

```diff
@@ -10,12 +11,15 @@ namespace Assets.Scripts.Runtime
     /// </summary>
     public class GameManager : MonoBehaviour
     {
+        [SerializeField] private WaveController _waveController;
```

Comment:
Wave controler doesnt make sense in the scope of game manager

The most common usecase for wave manager is:

- player enters the trigger in an arena part of the level
- door closes
- first wave of enemies spawn
- when those are killed, second wave spawns
- when second wave is klilled door opens 

and player can go to the next arena
where the sequence repeats

there can be more waves on a level, it is not a global thing - its local to a particular part of the map